platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/hash/VirtualHasher.java

```diff
@@ -436,29 +496,30 @@ public Hash hash(
                         continue;
                     }
                     if (siblingPath > lastLeafPath) {
+                        assert siblingPath == 2;
```

Comment:
I am not sure this is right. Assume chunk height is 3, there is only one chunk (starting off the root node), and leaf path range is 2-4. In this case, siblings for paths 5 and 6 would fall into this code branch, but the assertion is for path 2 only...

Reply:
It works because `parentRankHeights[lastLeafRank] == 1` so the only possibility for `siblingPath` to be greater than `lastLeafPath` is when it's a degenerate tree with `lastLeafPath == 1`.